app/controllers/application_controller.rb

```diff
@@ -58,4 +58,15 @@ def redirect_to_primary_host
 
     redirect_to [request.protocol, ENV["PRIMARY_HOST"], request.fullpath].join, allow_other_host: true
   end
+
+  def authenticate
+    return unless FeatureFlags.enabled?(:basic_authentication, without_session_data: true)
+
+    return true if authenticate_with_http_basic do |username, password|
```

Comment:
So reading [this ](https://stackoverflow.com/questions/5101020/in-ruby-on-rails-what-does-authenticate-with-http-basic-do) we can call `authenticate_or_request_with_http_basic` here without the need for `request_http_basic_authentication` - no idea if it'll work but thought it was worth a mention.

Reply:
Good spot! Yes, that looks like it rolls both things into one, which is much neater.